src/Options/Commander.hs

```diff
@@ -528,8 +529,17 @@ flag = FlagProgramT
 -- to run out of a program I want to write.
 toplevel :: forall s p m. (HasProgram p, KnownSymbol s, MonadIO m) 
          => ProgramT p m () 
-         -> ProgramT (Named s & ("help" & Raw + p)) m ()
-toplevel p = named (sub (usage @(Named s & ("help" & Raw + p))) <+> p)
+         -> ProgramT (Named s & ("help" & Raw + p + Raw)) m ()
+-- toplevel p = named (sub (usage @(Named s & ("help" & Raw + p))) <+> p)
+toplevel p = named $ helpSubprogram <+> p <+> helpIfMissed
+  where
+  helpSubprogram = sub $ usage @(Named s & ("help" & Raw + p))
+  helpIfMissed :: ProgramT Raw m ()
+  helpIfMissed = raw $ liftIO do
+    putStrLn "usage:"
+    putStrLn (document @p)
+    exitFailure
```

Comment:
Yeah, I totally agree with this. I think it means that the example can be simplified as well, please include that in this PR or explain why that is not the case, as a demonstration of the use of this function.

Reply:
Bump!

Reply:
could you comment on #42